[ Backported.  ]

commit 49f8d1c24c639d891f58a3b9feda425833b702fb
Author: H.J. Lu <hjl.tools@gmail.com>
Date:   Fri Mar 12 15:12:34 2010 +0000

    Restore sp for x86.
    
    2010-03-12  H.J. Lu  <hongjiu.lu@intel.com>
    
    	* amd64-tdep.c (amd64_word_names): Replace "sp" with "".
    	* i386-tdep.c (i386_word_names): Likewise.

--- gdb-7.1-p4/gdb/amd64-tdep.c	2010-04-03 21:06:52.000000000 +0200
+++ gdb-7.1/gdb/amd64-tdep.c	2010-04-03 21:08:21.000000000 +0200
@@ -222,7 +222,7 @@ static const char *amd64_byte_names[] =
 
 static const char *amd64_word_names[] =
 {
-  "ax", "bx", "cx", "dx", "si", "di", "bp", "sp", 
+  "ax", "bx", "cx", "dx", "si", "di", "bp", "", 
   "r8w", "r9w", "r10w", "r11w", "r12w", "r13w", "r14w", "r15w"
 };
 
--- gdb-7.1-p4/gdb/i386-tdep.c	2010-04-03 21:06:52.000000000 +0200
+++ gdb-7.1/gdb/i386-tdep.c	2010-04-03 21:08:21.000000000 +0200
@@ -94,7 +94,7 @@ static const char *i386_byte_names[] =
 static const char *i386_word_names[] =
 {
   "ax", "cx", "dx", "bx",
-  "sp", "bp", "si", "di"
+  "", "bp", "si", "di"
 };
 
 /* MMX register?  */
